dotcom-rendering/src/web/lib/liveblogAdSlots.ts

```diff
@@ -28,49 +38,57 @@ type BlockElementHeight = {
 
 /**
  * Approximations of the size each block element type will take up on the page in pixels.
- * Predictions are made for desktop, as we prefer to use a value closer to the lower bound,
- * as we would rather display too few ads than too many. Practically, this will mean that we
- * will show the the right frequency of ads on desktop and too few on smaller devices.
+ * Predictions are made for mobile viewports, as data suggests that the majority of liveblog page
```

Comment:
Just for my own understanding does this mean we optimise for mobile viewports and not necessarily for desktop? By doing so what is the impact on ad placement for desktop?

Reply:
We do optimise for mobile viewports. There are three options:

1, Optimise for mobile viewports

This may mean that we include too many ads on the page, as a mobile viewport height is smaller, however in testing of >100 pages, no desktop page had two ads in the same viewport on desktop. User device type is almost twice as likely to be mobile than it is desktop for liveblog pages, which is the key reason we went with this option.

2, Optimise for desktop viewports

This will mean that we will likely be rendering too few ads on mobile, since the slots are likely to be further apart. Whether we optimise for desktop or mobile, we can ensure that the overall ad-ratio remains consistent to the level it is currently.

3, Create slots for both, populating the only the relevant slots when the page loads, depending on the viewport

This adds more complexity and requires increasing the amount client-side code. However this would improve ad ratio and use a more "correct" amount of ads on both mobile and desktop. I hope we can AB test this in the future.

Reply:
Thanks for the explanation Dom 👍